### ISSUE TITLE
Bug: Bad/Missing SupportOSVersion on Windows.System.User.GetDefault

### ISSUE THREAD

**Issue by AliveDevil** [microsoft/CsWinRT#1844]
### Description

On [Windows.System.User.GetDefault method](https://learn.microsoft.com/uwp/api/windows.system.user.getdefault] it's stated that this is supported on `WinRT	Build 20348, Build 22000, Build 22621, Build 26100` only.
This is represented in the Windows.winmd file as well:
```
[ComImport]
[Windows.Foundation.Metadata.Guid(1956871697u, 11957, 17543, 176, 213, 44, 103, 144, 224, 19, 233)]
[ExclusiveTo(typeof(User))]
[ContractVersion(typeof(UniversalApiContract), 720896u)]
internal interface IUserStatics2
{
	[MethodImpl(MethodImplOptions.InternalCall, MethodCodeType = MethodCodeType.Runtime)]
	User GetDefault();
}
```

Assuming UniversalApiContract version 720896 is v12/10.0.20348.0.

This is a mismatch between the User-class and IUserStatics2 implemented interface:
User-class:
```
[MethodImpl(MethodImplOptions.InternalCall, MethodCodeType = MethodCodeType.Runtime)]
public static extern User GetDefault();
```

Must be correlated with IUserStatics2-interface (i.e. if IUserStatics2.GetDefault() is 10.0.20348.0 only, then User.GetDefault has the same restriction).

### Steps To Reproduce

Repro: [sdaujfg.zip](https://github.com/user-attachments/files/17506940/sdaujfg.zip)

I expect User.GetDefault() to be compiler warn me that this callsite is unavailable on TargetFrameworkMinVersion.

### Expected Behavior

GetDefault should be annotated with `SupportedOSVersion("10.0.20348.0")`, as is annotated on the IUserStatics2-interface in Windows metadata.

### Version Info

```
dotnet --info
.NET SDK:
 Version:           8.0.403
 Commit:            c64aa40a71
 Workload version:  8.0.400-manifests.e99c892e
 MSBuild version:   17.11.9+a69bbaaf5
```

```
<TargetFramework>net8.0-windows10.0.22621.0</TargetFramework>
<TargetFrameworkMinVersion>10.0.0.17763.0</TargetFrameworkMinVersion>
```

**Comment by dongle-the-gadget** [comment]
I don’t think we already have this for older frameworks. In addition, for system WinRT classes, version checking is discouraged to feature checking, which is much harder to guard.

**Comment by AliveDevil** [comment]
Fair, nevertheless, there is missing API contract information on User.GetDefault.
